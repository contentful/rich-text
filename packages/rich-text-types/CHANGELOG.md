# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [15.3.6](https://github.com/contentful/rich-text/compare/v15.3.5...v15.3.6) (2021-09-15)


### Bug Fixes

* allow table-row type to have both cells ([#267](https://github.com/contentful/rich-text/issues/267)) ([ec40598](https://github.com/contentful/rich-text/commit/ec405982109cb1c16e7adf71a541a98270d7f45b))





## [15.3.5](https://github.com/contentful/rich-text/compare/v15.3.4...v15.3.5) (2021-09-13)


### Bug Fixes

* **rich-text-types:** forbid Tables inside ListItem ([#266](https://github.com/contentful/rich-text/issues/266)) ([fc338bf](https://github.com/contentful/rich-text/commit/fc338bf040b8718057717d2681f800d5e26ba59d))





## [15.3.3](https://github.com/contentful/rich-text/compare/v15.3.2...v15.3.3) (2021-09-07)

**Note:** Version bump only for package @contentful/rich-text-types





## [15.3.2](https://github.com/contentful/rich-text/compare/v15.3.1...v15.3.2) (2021-09-07)

**Note:** Version bump only for package @contentful/rich-text-types





## [15.3.1](https://github.com/contentful/rich-text/compare/v15.3.0...v15.3.1) (2021-09-07)

**Note:** Version bump only for package @contentful/rich-text-types





# [15.3.0](https://github.com/contentful/rich-text/compare/v15.2.0...v15.3.0) (2021-09-06)


### Bug Fixes

* allow the first item in Table to be a TableHeaderRow ([9a6de55](https://github.com/contentful/rich-text/commit/9a6de55ed97ca413d3922958850b8215922c06b5))
* don't export the helper TableHeaderRow type ([179c3cb](https://github.com/contentful/rich-text/commit/179c3cb10c0f65725297445afe955f0cca135005))
* typo ([2282665](https://github.com/contentful/rich-text/commit/2282665924404771b3353dd9b380d8863f1d4c41))


### Features

* add TableHeaderCell type ([5e25ac9](https://github.com/contentful/rich-text/commit/5e25ac9f35be2ad6c7ad7857cbde808cbc3437f9))


# [15.2.0](https://github.com/contentful/rich-text/compare/v15.2.0...v15.1.0) (2021-09-06)


### Bug Fixes

* allow the first item in Table to be a TableHeaderRow ([9a6de55](https://github.com/contentful/rich-text/commit/9a6de55ed97ca413d3922958850b8215922c06b5))
* don't export the helper TableHeaderRow type ([179c3cb](https://github.com/contentful/rich-text/commit/179c3cb10c0f65725297445afe955f0cca135005))
* typo ([2282665](https://github.com/contentful/rich-text/commit/2282665924404771b3353dd9b380d8863f1d4c41))


### Features

* add TableHeaderCell type ([5e25ac9](https://github.com/contentful/rich-text/commit/5e25ac9f35be2ad6c7ad7857cbde808cbc3437f9))



# [15.1.0](https://github.com/contentful/rich-text/compare/v15.0.0...v15.1.0) (2021-08-02)


### Features

* add rowspan ([e0264fb](https://github.com/contentful/rich-text/commit/e0264fbe4d9467eedf78b7c5c3a9825a584124cf))



# [15.0.0](https://github.com/contentful/rich-text/compare/v14.2.0...v15.0.0) (2021-06-15)


### Features

* add table markup support ([c6ae127](https://github.com/contentful/rich-text/commit/c6ae127aa460f4cd26cd6b671cd43fd2714bc650))



## [14.1.2](https://github.com/contentful/rich-text/compare/v14.0.1...v14.1.2) (2020-11-02)


### Bug Fixes

* üêõ configure rollup copy correctly ([3446a33](https://github.com/contentful/rich-text/commit/3446a33fc5711087095a58b088828dfe6066bc7f))



## [14.0.1](https://github.com/contentful/rich-text/compare/v14.0.0...v14.0.1) (2020-01-30)


### Features

* üé∏ emptyDoc representing an empty RT document ([80315ef](https://github.com/contentful/rich-text/commit/80315ef5130b79336336ba31de8a55e42bafe319))



# [14.0.0](https://github.com/contentful/rich-text/compare/v13.4.0...v14.0.0) (2020-01-28)



# [13.4.0](https://github.com/contentful/rich-text/compare/v13.3.0...v13.4.0) (2019-08-01)



# [13.1.0](https://github.com/contentful/rich-text/compare/v13.0.1...v13.1.0) (2019-03-04)



# [13.0.0](https://github.com/contentful/rich-text/compare/v12.2.1...v13.0.0) (2019-01-22)



## [12.1.2](https://github.com/contentful/rich-text/compare/v12.1.1...v12.1.2) (2018-12-14)



## [12.0.1](https://github.com/contentful/rich-text/compare/v12.0.0...v12.0.1) (2018-12-04)


### Bug Fixes

* üêõ add Object.values and Array.prototype.includes polyfills ([9e2ffb4](https://github.com/contentful/rich-text/commit/9e2ffb46e3564a523c74504270b29ccc8b8249ad))



# [12.0.0](https://github.com/contentful/rich-text/compare/v11.0.0...v12.0.0) (2018-11-29)



# [11.0.0](https://github.com/contentful/rich-text/compare/v10.3.0...v11.0.0) (2018-11-27)


### Bug Fixes

* üêõ Removes obsolete field `title` from Hyperlinks ([8cb9027](https://github.com/contentful/rich-text/commit/8cb90279f5348a7fb59f211c2ba209a28fd432be))



# [10.3.0](https://github.com/contentful/rich-text/compare/v10.2.0...v10.3.0) (2018-11-26)



# [10.2.0](https://github.com/contentful/rich-text/compare/v10.1.0...v10.2.0) (2018-11-19)


### Features

* üé∏ Add rich-text-references ([363b4e5](https://github.com/contentful/rich-text/commit/363b4e509e94af0932fd7cece8e56beafe8d67d2))



# [10.1.0](https://github.com/contentful/rich-text/compare/v10.0.5...v10.1.0) (2018-11-16)


### Features

* üé∏ adds json schema generation to rich text types ([8916140](https://github.com/contentful/rich-text/commit/89161404eb911f126be23ba2a146ebf748f7489e))
* üé∏ introduces top-level-block type ([a6bf35e](https://github.com/contentful/rich-text/commit/a6bf35e7c9ca35915a512de774b3a3fdc4c76e5d))



## [10.0.1](https://github.com/contentful/rich-text/compare/v10.0.0...v10.0.1) (2018-11-08)



# [10.0.0](https://github.com/contentful/rich-text/compare/v9.0.2...v10.0.0) (2018-11-02)


### Features

* üé∏ removes nodeClass from Node interface ([09b8162](https://github.com/contentful/rich-text/commit/09b8162bcba65bc13afa14b2b5ff046c9fed7b3b))


### BREAKING CHANGES

* Removes accidentally added nodeClass



## [9.0.2](https://github.com/contentful/rich-text/compare/v9.0.1...v9.0.2) (2018-10-31)



# [9.0.0](https://github.com/contentful/rich-text/compare/v8.0.3...v9.0.0) (2018-10-30)


### Features

* üé∏ Explicitly declare nodeClass and nodeType in core types ([0749c61](https://github.com/contentful/rich-text/commit/0749c6199bb2681509608539c76bd8149cade564))


### BREAKING CHANGES

* Potentially breaks TypeScript libraries pulling this in as a dependency
if they are not explicitly stating nodeClass and nodeType.



## [8.0.3](https://github.com/contentful/rich-text/compare/v8.0.2...v8.0.3) (2018-10-30)



## [8.0.2](https://github.com/contentful/rich-text/compare/v8.0.1...v8.0.2) (2018-10-29)



## [8.0.1](https://github.com/contentful/rich-text/compare/v8.0.0...v8.0.1) (2018-10-26)


### Bug Fixes

* üêõ Revert mock hypenation commits ([bde9432](https://github.com/contentful/rich-text/commit/bde94323fcc02bf5ab3feeef46a9d8fc8b08d59b))


### Features

* üé∏ take all packages out of "demo mode" ([815f18b](https://github.com/contentful/rich-text/commit/815f18be6a914e7e4782790ee46053689712494b))


### BREAKING CHANGES

* renames all packages



# 6.0.0 (2018-10-24)
