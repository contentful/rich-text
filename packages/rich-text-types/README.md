# rich-text-types

Type definitions and constants for the Contentful rich text field type.
