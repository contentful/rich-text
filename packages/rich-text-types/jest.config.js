/* eslint-disable */

const getBaseConfig = require('../../baseJestConfig');
const packageJson = require('./package.json');
const packageName = packageJson.name.split('@contentful/')[1];

module.exports = {
  ...getBaseConfig(packageName),
};
