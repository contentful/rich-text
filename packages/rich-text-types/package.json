{
  "name": "@contentful/rich-text-types",
  "version": "15.15.1",
  "main": "dist/index.js",
  "typings": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/contentful/rich-text.git"
  },
  "license": "MIT",
  "engines": {
    "node": ">=6.0.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "npm run generate-json-schema && tsc --module commonjs && rollup -c rollup.config.js",
    "start": "tsc && rollup -c rollup.config.js -w",
    "lint": "tslint -t codeFrame '@(src|bin)/*.ts'",
    "generate-json-schema": "ts-node -O '{\"module\": \"commonjs\"}' ./tools/jsonSchemaGen"
  },
  "devDependencies": {
    "@cspotcode/source-map-consumer": "^0.8.0",
    "@types/jest": "^27.0.1",
    "@types/node": "^14.17.14",
    "faker": "^4.1.0",
    "jest": "^27.1.0",
    "rimraf": "^2.6.3",
    "rollup": "^1.32.1",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-copy": "^2.0.0",
    "rollup-plugin-json": "^4.0.0",
    "rollup-plugin-node-resolve": "^4.2.4",
    "rollup-plugin-sourcemaps": "^0.6.3",
    "rollup-plugin-typescript2": "^0.30.0",
    "ts-jest": "^27.0.5",
    "ts-node": "^10.2.1",
    "tslib": "^2.3.1",
    "tslint": "^6.1.3",
    "typescript": "^4.4.2",
    "typescript-json-schema": "^0.50.1"
  },
  "gitHead": "c7e34bc0f0172728999b121af43df88995eec305"
}
