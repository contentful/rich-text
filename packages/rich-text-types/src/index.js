"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var blocks_1 = require("./blocks");
exports.BLOCKS = blocks_1.default;
var inlines_1 = require("./inlines");
exports.INLINES = inlines_1.default;
var marks_1 = require("./marks");
exports.MARKS = marks_1.default;
__export(require("./schemaConstraints"));
//# sourceMappingURL=index.js.map