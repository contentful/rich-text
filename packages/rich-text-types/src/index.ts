export { BLOCKS } from './blocks';
export { INLINES } from './inlines';
export { MARKS } from './marks';

export * from './schemaConstraints';

export * from './types';
export * from './nodeTypes';

export { EMPTY_DOCUMENT } from './emptyDocument';

import * as helpers from './helpers';
export { helpers };

export { getSchemaWithNodeType } from './schemas';
