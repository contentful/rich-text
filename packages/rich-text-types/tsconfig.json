{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declarationDir": "dist/types",
    "outDir": "dist",
    "typeRoots": ["../../node_modules/@types", "node_modules/@types", "src/typings"],
    "types": ["jest", "node"],
    "inlineSources": true,
  },
  "include": ["src"]
}
