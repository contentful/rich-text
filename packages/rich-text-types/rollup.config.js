import copy from 'rollup-plugin-copy';

import config from '../../rollup.config';
import { main as outputFile } from './package.json';

const customConfig = () => {
  const baseConfig = config(outputFile);

  return {
    ...baseConfig,
    output: {
      // Maintain the same file names & folders structure as "src"
      // This is necessary to resolve generated JSON schema files
      //
      // https://rollupjs.org/guide/en/#outputpreservemodules
      preserveModules: true,
    },
    plugins: [
      ...baseConfig.plugins,
      copy({
        targets: [
          {
            src: 'src/schemas/generated',
            dest: 'dist/schemas',
          },
        ],
      }),
    ],
  };
};

export default customConfig();
