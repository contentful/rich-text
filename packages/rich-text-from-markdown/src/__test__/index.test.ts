import { richTextFromMarkdown } from '..';
import { Text, Block } from '@contentful/rich-text-types';

describe('rich-text-from-markdown', () => {
  test('should render some markdown', async () => {
    const document = await richTextFromMarkdown('# Hello World');
    const text = document.content[0].content[0] as Text;
    expect(text.value).toBe('Hello World');
  });

  test('should call the fallback function when a node is not supported', async () => {
    const fakeNode = { nodeType: 'image' };
    const fallback = jest.fn(node => Promise.resolve(fakeNode));
    const document = await richTextFromMarkdown(
      '![image](https://image.example.com/image.jpg)',
      fallback,
    );
    expect(document.content[0].nodeType).toBe('image');
    expect(fallback).toBeCalledTimes(1);
    expect(fallback.mock.calls).toMatchSnapshot();
  });
});

describe.each([
  ['*This is an italic text*', 'This is an italic text'],
  ['__This a bold text__', 'This a bold text'],
  ['`This is code`', 'This is code'],
])('The markdown "%s" should be parsed to text with value "%s"', (markdown, expected) => {
  test(`${expected}`, async () => {
    const document = await richTextFromMarkdown(markdown);
    const parsedText = document.content[0].content[0] as Text;
    expect(parsedText.value).toBe(expected);
  });
});

describe('parses complex inline image markdown correctly', () => {
  test('incoming markdown tree calls fallback twice', async () => {
    const fakeNode = { nodeType: 'image' };
    const fallback = jest.fn(node => Promise.resolve(fakeNode));
    const document = await richTextFromMarkdown(
      `![image](https://image.example.com/image.jpg) 
      ![image](https://image.example.com/image2.jpg)`,
      fallback,
    );
    expect(document.content.length).toBe(3);
    expect(document.content[0].nodeType).toBe('image');
    expect(document.content[1].nodeType).toBe('paragraph');
    expect(document.content[1].content[0].nodeType).toBe('text');
    expect(document.content[2].nodeType).toBe('image');
    expect(fallback).toBeCalledTimes(2);
    expect(fallback.mock.calls).toMatchSnapshot();
  });
  test('incoming markdown tree calls fallback twice', async () => {
    const fakeNode = { nodeType: 'image' };
    const fallback = jest.fn(node => Promise.resolve(fakeNode));
    const document = await richTextFromMarkdown(
      `some text ![image](https://image.example.com/image.jpg)![image](https://image.example.com/image2.jpg) some more text`,
      fallback,
    );
    expect(document.content.length).toBe(4);
    expect(document.content[0].nodeType).toBe('paragraph');
    expect(document.content[0].content[0].nodeType).toBe('text');
    expect(document.content[1].nodeType).toBe('image');
    expect(document.content[2].nodeType).toBe('image');
    expect(document.content[3].nodeType).toBe('paragraph');
    expect(fallback).toBeCalledTimes(2);
    expect(fallback.mock.calls).toMatchSnapshot();
  });
});
