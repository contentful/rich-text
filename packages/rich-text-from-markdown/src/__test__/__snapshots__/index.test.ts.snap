// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses complex inline image markdown correctly incoming markdown tree calls fallback twice 1`] = `
Array [
  Array [
    Object {
      "alt": "image",
      "position": Position {
        "end": Object {
          "column": 46,
          "line": 1,
          "offset": 45,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "title": null,
      "type": "image",
      "url": "https://image.example.com/image.jpg",
    },
  ],
  Array [
    Object {
      "alt": "image",
      "position": Position {
        "end": Object {
          "column": 53,
          "line": 2,
          "offset": 99,
        },
        "indent": Array [],
        "start": Object {
          "column": 7,
          "line": 2,
          "offset": 53,
        },
      },
      "title": null,
      "type": "image",
      "url": "https://image.example.com/image2.jpg",
    },
  ],
]
`;

exports[`parses complex inline image markdown correctly incoming markdown tree calls fallback twice 2`] = `
Array [
  Array [
    Object {
      "alt": "image",
      "position": Position {
        "end": Object {
          "column": 56,
          "line": 1,
          "offset": 55,
        },
        "indent": Array [],
        "start": Object {
          "column": 11,
          "line": 1,
          "offset": 10,
        },
      },
      "title": null,
      "type": "image",
      "url": "https://image.example.com/image.jpg",
    },
  ],
  Array [
    Object {
      "alt": "image",
      "position": Position {
        "end": Object {
          "column": 102,
          "line": 1,
          "offset": 101,
        },
        "indent": Array [],
        "start": Object {
          "column": 56,
          "line": 1,
          "offset": 55,
        },
      },
      "title": null,
      "type": "image",
      "url": "https://image.example.com/image2.jpg",
    },
  ],
]
`;

exports[`rich-text-from-markdown should call the fallback function when a node is not supported 1`] = `
Array [
  Array [
    Object {
      "alt": "image",
      "position": Position {
        "end": Object {
          "column": 46,
          "line": 1,
          "offset": 45,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "title": null,
      "type": "image",
      "url": "https://image.example.com/image.jpg",
    },
  ],
]
`;
